python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -194,6 +201,7 @@ def run(self):
         finally:
             # Signal end of results.
             self._output_node.outqueue.append(None)
+            _remove_metrics({"dataset": self._dataset_uuid})
```

Comment:
can we leave a comment here explaining why we need this? does this ensure the metrics don't persist on the dashboard after dataset execution completes?

Reply:
I'm not sure how it works under the hood, but you won't see the metrics for a dataset if there are no non-zero datapoints in the current time window

Reply:
what's the high level goal of this?

we don't want to see a line for a dataset if there are no values for that dataset?

That should already be the case since you are passing the dataset uuid into each of the metrics. Any uuid that hasn't used in a metric yet will not produce a line.

Reply:
Or is it, that at the completion of a dataset, you want the values to reset to 0?

If it's this one, I think renaming "remove_metrics" to "reset_metrics" or "clear_metrics" makes more sense as a name.

Reply:
Yes, it's to reset the metric after execution to avoid something like this:![Screenshot 2023-10-06 at 2 56 49 PM](https://github.com/ray-project/ray/assets/39287272/b3852f87-4c8f-4f57-9b1f-f6e948dff9aa)
This dataset finishes execution at around 10:10, and if we don't reset, it'll keep emitting it's last value. I'll rename this to make it more clear.